indra/newview/llfloaterimnearbychat.cpp

```diff
@@ -606,6 +608,13 @@ void LLFloaterIMNearbyChat::sendChat( EChatType type )
                 if(!LLGestureMgr::instance().triggerAndReviseString(utf8text, &utf8_revised_text))
                 {
                     utf8_revised_text = utf8text;
+                    // check if the message is /filename.lua and execute the Lua script
+                    static const boost::regex is_lua_script("^/.*\\.luau?(?:\\s+\\S+)*$");
```

Comment:
Wouldn't it be equivalent (and more readable) just to end the pattern string at `"...luau?"`

Reply:
Currently it matches strings like `/filename.lua` or `/filename.lua arg1 arg2`. Changing pattern to "^/.*\\.luau?" will allow matches like `/filename.luaqwerty123'.

Reply:
Changing pattern to `^/.*\\.luau?` won't allow matches like  `/filename.lua arg1 arg2`.

Reply:
Even if you omit the final `$`? I thought that would match any string prefix.

You're right about trailing characters though. You could use `"^/.*\\.luau?\\b"` ?